### PR TITLE
Show predefined string for promise's value

### PR DESCRIPTION
**Pull request by JeongPyoHam**

### CODE REVIEW

File: src/R/Editor/Impl/Data/RSessionDataObject.cs [hunk 1]
```diff
@@ -57,7 +57,7 @@ public class RSessionDataObject : IRSessionDataObject {
                 const string PromiseVaue = "<promise>";
```
Comment (huguesv) [inline src/R/Editor/Impl/Data/RSessionDataObject.cs]: Might as well fix that typo while you're here.

**Comment by huguesv** [comment]
:+1:

**Comment by int19h** [comment]
Why not show the promise expression? It's more informative, and it is also what RS does.

**Comment by huguesv** [comment]
It creates an unreadable wall of text.  Is the lost information useful in any way?

**Comment by int19h** [comment]
It depends on what it is. In many cases, it can tell you what the value is, before it becomes evaluated. E.g. suppose you have this:
```r
f <- function(x) {
 ...
}
f(123)
```
and you have a breakpoint set at the first line of the function. Without this change, you'll see `x` show `123` at the breakpoint, but with it, you'll see `<promise>`.

Can you give a specific scenario where it results in a wall of text? We probably just need to do some trimming.

**Comment by huguesv** [comment]
image

Maybe only special case lazyLoadDBfetch?

File: src/R/Editor/Impl/Data/RSessionDataObject.cs (hunk 1 above)
Comment (JeongPyoHam) [inline src/R/Editor/Impl/Data/RSessionDataObject.cs]: Ah! fixed.

**Comment by int19h** [comment]
It appears that RS changes behavior depending on whether you're viewing a function environment, or a package environment. In function environment, promise code is displayed. In package environment, it just says `<Promise>`. Perhaps we should be doing the same.

**Comment by int19h** [comment]
But yes, since `lazyLoadDBfetch` appears to be something that the package loader uses to initialize a lot of those, special-casing it would appear to be a viable approach, as well. Basically if it's something else, it's probably interesting.

**Comment by JeongPyoHam** [comment]
In the tiny cell, function body, which is cut off, doesn't show more than just 'function', and hard to be useful. When code is required in that cell, new issue will be opened.

**Comment by int19h** [comment]
This is not just about the function bodies, though. This is about promises for function arguments, which are more often than not simple expressions that are helpful, as in my example above.

**Comment by JeongPyoHam** [comment]
As long as it shows more than just <promise>, you can use the tiny part for leverage to your justification. The claimed usefulness is limited to trivial cases where representation fits into the tiny cell.
This is for addressing the specific issue of clearing up package's list, and one change can't contain everything that appears in mind randomly. The concern of showing really useful information in the mentioned area should be addressed in bigger scope such as triggering View with double click or hovering mosue shows code or expanding the cell to bigger one at click.
If even such small snippet required, there will be feedback.

**Comment by int19h** [comment]
My concern is that this change is addressing the specific issue of clearing up package list (#1370) by changing the behavior globally, such that it affects way more than packages; and the resulting change of behavior actually makes it different from RS when it was identical before (for function arguments).

**Pull request merged**